### ISSUE TITLE
Workflow orchard multiple Events with the same type

### ISSUE THREAD

**Issue by ghost** [OrchardCMS/Orchard#8006]
![WorkFlow](https://user-images.githubusercontent.com/22644374/38136105-97972ba2-341c-11e8-92f2-c9fb0b5e63ad.PNG)

as you can see a file loaded event is trigged once a file has been uploaded by the admin then it notifies a specific role (E.g role1) then it will wait until that role accept or decline the file . once he accept it it will trigger notifyRole now on role2 then it will wait until that role accept or decline the file again ...

Now the notifyRole works perfectly it notify the specified role (on the Form of the activity) . 
but the Event (StatusChanged) i make a condition something like if the currentRole of the file example role1 and you are the StatusEvent which listens for the role1 then do something ... I'm also passing the Role using Form on StatusChanged Edit activity . 

the issue : if another role Say Role2 Accept the file its fires the first event always but the weird thing here is it ends the workflow because its notify the last role which is Role3 even tho the First Event was fired . 

is there anyway to accomplish such workflow ? 

any help is really appreciated.

**Comment by sebastienros** [comment]
Looks like a bug, we should check that the WF is waiting on a previous task.

**Comment by ghost** [comment]
@sebastienros  yes it also does not pass the tokens from the previous activity to the event it only pass tokens that were sent by the event itself

**Comment by sfmskywalker** [comment]
I tested this scenario extensively in OC. No problems there.

**Comment by ghost** [comment]
@sfmskywalker  I got to save the current (notify Role) activity's destination position and in the StatusChanged Event  I check if the current positions equals to the saved position .

**Comment by sfmskywalker** [comment]
You mean in O1, or OC? In OC, you shouldn't have to do that - the workflow manager will do the right thing when resuming a workflow instance. If not, let me know.

**Comment by ghost** [comment]
@sfmskywalker  no in O1 (1.10.2)
